clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
Is this the mechanism by which resolver implementations will return service config data to the grpc client code?  If so, then I think this should return the service config in JSON form, as per previous design discussions.

Also, we should have an API for returning the service config in JSON form to the application, for end-to-end debugging purposes.